aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -236,7 +236,9 @@ std::tuple<Tensor, Tensor, double, int64_t> fbgemm_linear_quantize_weight(
   q_params.precision = kPrecision;
 
   Tensor quantized = at::native::empty_like(
-      weight_contig, weight_contig.options().dtype(at::kChar), LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+      weight_contig, at::kChar, weight_contig.options().layout(), 
```

Comment:
is the issue here that we don't generate a TensorOptions based native function?

Reply:
(see also my comment in the main issue body)

Reply:
Im not sure if i understand the question. I dont change any logic here, just "unwrap" TensorOptions to 4 params. Behavior isn't changed.

Reply:
I think gchanan is referring to the fact that you had to replace a simple call with a more complex, error prone call with all of the four parameters. The fact that you had to do that in the first place is a code small.